hypothesis-python/src/hypothesis/extra/numpy.py

```diff
@@ -213,6 +213,8 @@ def compat_kw(*args, **kw):
         else:  # NEP-7 defines the NaT value as integer -(2**63)
             elems = st.integers(-(2**63) + 1, 2**63 - 1)
         result = st.builds(dtype.type, elems, res)
+    elif dtype.kind == "O":
+        return st.from_type(object)
     else:
```

Comment:
it's not actually clear to me whether we want `st.from_type(object)` or `from_type(type).flatmap(st.from_type)` here. Should we make the former simply register to the latter?